### PR TITLE
SAFROLE fixes/refactor

### PR DESCRIPTION
**Pull request by greywolve**
The primary goal here is to handle a new epoch change with SAFROLE. In this case the old state alone is not sufficient, and we need to figure out what the relevant components in the state will become in order to properly produce blocks, submit tickets, and determine slot leaders.

- Refactor sealing and ticket functions to rather accept exactly what they need rather then the entire state, this makes it simpler to unit test them and also allows us to push more complex logic to handle the epoch change edge cases out to a few core node functions that will eventually produce blocks, submit tickets and determine slot leaders.
- Refactor RingVerify/Prover to rather just operate on ValidatorsData directly since we can't just rely on the current state.
- Rename the TicketAccumulator to SealingKeys to avoid confusion.
- Update MaxTicketExtrinsicSize to 3 for tiny constants.
- Update the simulation test to properly handle the epoch change edge case.

### CODE REVIEW

File: internal/state/ticket.go [hunk 1]
```diff
@@ -40,9 +43,9 @@ func CreateTicketProof(state *State, privateKey crypto.BandersnatchPrivateKey, a
 // Verifies a ticket proof as per equation 6.29 in the graypaper (v0.6.4)
 // Returns the output hash / identifier of the ticket if the proof is valid, or
 // an error if it is not.
-func VerifyTicketProof(state *State, ticket block.TicketProof) (crypto.BandersnatchOutputHash, error) {
+func VerifyTicketProof(ringCommitment crypto.RingCommitment, entropy crypto.Hash, ticket block.TicketProof) (crypto.BandersnatchOutputHash, error) {
 	// TODO: if this is too expensive to construct each time we should pass it in.
-	ringVerifier, err := state.ValidatorState.SafroleState.RingVerifier()
+	ringVerifier, err := bandersnatch.NewRingVerifier([]crypto.BandersnatchPublicKey{})
```
[the remark this answers is not in the document]
Comment (greywolve) [reply internal/state/ticket.go:48]: Actually this can be empty. It's something I realized when I was refactoring. When we do the ring signature verification we use the ring commitment, and hence we don't actually need to construct a full ring to verify. (It's a shortcut to prevent having to compute the commitment each time which is expensive) 

The bandersnatch VRFS rust example did it this way, but it could be simplified to a function instead of needing to construct an object. We'd only need to construct the object for computing the ring commitment.

I could add a TODO here or above the function itself, I'd prefer to do that refactor later because I need to modify the rust lib code too.

[the remark this answers is not in the document]
Comment (greywolve) [reply internal/state/ticket.go]: Added the TODO and more comments.

**Review by bamzedev — DISMISSED** [review]

**Review by pantrif — APPROVED** [review]
